src/pyenphase/envoy.py

```diff
@@ -310,17 +311,31 @@ async def open_dry_contact(self, id: str) -> dict[str, Any]:
                 "This feature is not available on this Envoy."
             )
 
-        return await self._json_request(
-            URL_DRY_CONTACT_STATUS, {"dry_contacts": {"id": id, "status": "open"}}
+        result = await self._json_request(
+            URL_DRY_CONTACT_STATUS,
+            {"dry_contacts": {"id": id, "status": DryContactStatus.OPEN}},
         )
 
+        # The envoy returns the old state for a few seconds after the relay is changed, so we pre-emptively update
+        # the status in our data
+        if self.data:
```

Comment:
Maybe assert `self.data` is not `None` at the top of the function since it never should be